### PR TITLE
Add new Panoptes client, and set a default Panoptes environment

### PR DESCRIPTION
**Pull request by rogerhutchings**
`panoptes-javascript-client` uses process.env.NODE_ENV to determine which API urls to use. `development` doesn't match any of its known options, so it will error out (specifically on SugarClient) if it doesn't have a valid one set. This configures webpack to define the environment as production in order to hit the live API, which I'm guessing people will probably want as a default.

### CODE REVIEW

**Comment by rogerhutchings** [comment]
After talking about this with @brian-c, I've decided to change this. Setting the `NODE_ENV` to `production` can suppress some useful debug information from React, plus he rightly pointed out that we probably shouldn't be setting env vars for people.

However: `panoptes-javascript-client` still requires a valid development to be set, as it uses it to set which API to hit (staging, API, whatever). Also, I really don't think your Node environment should determine which endpoint you're working with - you might want to develop locally against the production API, but that would mean switching _everything_ to production.

Fortunately, you can override the `panoptes-javascript-client` using a query parameter in the URL. So my minimally invasive approach will be to:

- Set the NODE_ENV in webpack from the actual Node environment
- Configure the Express dev server to rewrite the URL to include a Panoptes environment override (defaulting to production)

This will result in a slightly ugly URL for local dev purposes (e.g. `http://localhost:3000/?env=production#/?_k=b7hbjx`), but that's not a lot to live with.

In future, it may be worth decoupling the `panoptes-javascript-client` environment from the `NODE_ENV`, by using `process.env.PANOPTES_ENV` property to configure the API endpoint, for example.

File: server.js [hunk 1]
```diff
@@ -26,12 +26,24 @@ if (!isProduction) {
     },
   });
 
+  function setPanoptesEnvMiddleware(req, res, next) {
+    if (req.url === '/' && !req.url.match(/\W?env=(\w+)/) {
```
Comment (brian-c) [inline server.js]: Am I reading this right? If `req.url === '/'`, then `req.url` will never `.match(/\W?env=(\w+)/)`.

And is it missing a closing `)`?

File: server.js [hunk 2]
```diff
@@ -43,6 +55,6 @@ app.listen(port, '0.0.0.0', function onStart(err) {
   if (err) {
     console.log(err);
   }
-
+  console.info('Node environment is set to: %s', process.env.NODE_ENV);
   console.info('==> Listening on port %s. Open up http://0.0.0.0:%s/ in your browser.', port, port);
```
Comment (brian-c) [inline server.js]: I don't actually think 0.0.0.0 is allowed on the back end. Switch to "localhost"?

File: server.js (hunk 1 above)
Comment (rogerhutchings) [inline server.js]: Right on both counts - it was working in spite of me. Fixed to match only `/`

**Comment by rogerhutchings** [comment]
...now when I come to demonstrate the bug that made me do all this in the first place, I can't replicate it. FML. closing

**Pull request closed without merging**